### DISCUSSION TITLE
Scanner

### DISCUSSION THREAD

**Discussion by DerryLin-Tamu** [SimDynamX/SpaceTeamsPRO#73] [2024 Lunar SAR Competition Q&A] [answered]
How does the scanner work? In the GetTargetScanStatus(en) function it checks scanner and camera scanner. Is there specs to the scanner (range, needs to be in frame on camera)?

**Comment by connorjak** [comment, marked as answer]
The rovers have a Scanner and a CamScanner. These are both also entities in the sim. You can look for the `ScanRadius_m` parameter in the sim config to see their specs.

Scanner ("LIDAR") has a spherical pattern with the radius.

CamScanner (Camera search volume) has a conical pattern (approximated as a capsule/cylinder), where the radius is the max distance the pattern reaches and 2x `TraceConeAngle_deg` = camera field of view. The point of the cone is at the camera location. CamScanner is attached to the camera, so using `CameraPan` command will rotate the cam scanner.
